core/src/main/java/org/apache/calcite/rel/rules/ProjectMergeRule.java

```diff
@@ -54,8 +55,8 @@
    */
   public ProjectMergeRule(boolean force, RelBuilderFactory relBuilderFactory) {
     super(
-        operand(Project.class,
-            operand(Project.class, any())),
+        operand(LogicalProject.class,
+            operand(LogicalProject.class, any())),
```

Comment:
The changes sounds reasonable to me, but i'm a little worried about the compatibility. It would give us more confidence if some engine like Druid can give feedback that this change does not break something

Reply:
Do you know who is the right contact for Druid?

Reply:
I think @zabetak can make some help ~